src/formatDate.js

```diff
@@ -8,7 +8,31 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObject = {};
+  const separator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const splitedDate = date.split(separator);
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = splitedDate[i];
+  }
+
+  if (dateObject.hasOwnProperty('YYYY')) {
+    dateObject.YY = dateObject.YYYY.slice(2);
+  }
+
+  if (dateObject.YY >= 30) {
+    dateObject.YYYY = '19' + dateObject.YY;
+  } else {
+    dateObject.YYYY = '20' + dateObject.YY;
+  }
```

Comment:
The logic for converting 'YY' to 'YYYY' is always executed, regardless of whether 'YY' is present in the 'dateObject'. This could lead to incorrect behavior if 'YY' is not part of the input 'fromFormat'. You should only perform this conversion if 'YY' is a property of 'dateObject'.